### PR TITLE
Add an inference example for wavernn

### PR DESCRIPTION
**Pull request by yangarbiter**
This example takes in an example from LJSPEECH and transform it into a mel spectrogram. The inference function of WaveRNN should be able to transform the mel spectrogram back to it's original waveform.

The implementation if ported from [here](https://github.com/fatchord/WaveRNN/blob/master/models/fatchord_version.py#L169).

@mthrok we may want to discuss whether to put [the inference function](https://github.com/pytorch/audio/pull/1637/files#diff-2f1a2a7054ae6468415baac0a5a6c4d7e40a36350942d65c8ba0ba3b469c47a4R181) into the core library since in the current form, it is not torchscriptable.

### CODE REVIEW

**Comment by mthrok** [comment]
The summary of offline discussion
---

> @mthrok we may want to discuss whether to put [the inference function](https://github.com/pytorch/audio/pull/1637/files#diff-2f1a2a7054ae6468415baac0a5a6c4d7e40a36350942d65c8ba0ba3b469c47a4R181) into the core library

Extract the model data handling part from the current `infer` function to see if we can make the proposed `infer` method simple and minimalistic. (Things like mulaw and batch folding should be handled outside of model method)

**Comment by yangarbiter** [comment]
> The summary of offline discussion
> 
> > @mthrok we may want to discuss whether to put [the inference function](https://github.com/pytorch/audio/pull/1637/files#diff-2f1a2a7054ae6468415baac0a5a6c4d7e40a36350942d65c8ba0ba3b469c47a4R181) into the core library
> 
> Extract the model data handling part from the current `infer` function to see if we can make the proposed `infer` method simple and minimalistic. (Things like mulaw and batch folding should be handled outside of model method)

If appears that the inference function is not yet jitable due to [`torch.distributions.Categorical`](https://github.com/pytorch/audio/pull/1637/files#diff-2f1a2a7054ae6468415baac0a5a6c4d7e40a36350942d65c8ba0ba3b469c47a4R248) being unjitable (https://github.com/pytorch/pytorch/issues/18094)

So for future improvement, we want to move 
1. mulaw decoding,
2. batch folding, and
3. categorical sampling
outside the inference method

**Comment by mthrok** [comment]
> If appears that the inference function is not yet jitable due to [`torch.distributions.Categorical`](https://github.com/pytorch/audio/pull/1637/files#diff-2f1a2a7054ae6468415baac0a5a6c4d7e40a36350942d65c8ba0ba3b469c47a4R248) being unjitable ([pytorch/pytorch#18094](https://github.com/pytorch/pytorch/issues/18094))
> 

The document says `torch.distributions.Categorical` is equivalent to `torch.multinomial`. Can you try that?

**Comment by yangarbiter** [comment]
> > If appears that the inference function is not yet jitable due to [`torch.distributions.Categorical`](https://github.com/pytorch/audio/pull/1637/files#diff-2f1a2a7054ae6468415baac0a5a6c4d7e40a36350942d65c8ba0ba3b469c47a4R248) being unjitable ([pytorch/pytorch#18094](https://github.com/pytorch/pytorch/issues/18094))
> 
> The document says `torch.distributions.Categorical` is equivalent to `torch.multinomial`. Can you try that?

Thanks a lot for the suggestion! It works!

File: examples/pipeline_wavernn/wavernn_inference_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+            padded[:, :, pad:pad + t] = x
+        elif side == 'after':
+            padded[:, :, :t] = x
+        else:
+            raise ValueError(f"Unexpected side: '{side}'. "
+                            f"Valid choices are 'both', 'before' and 'after'.")
+        return padded
+
+    @torch.jit.export
+    def infer(self,
```
Comment (mthrok) [inline examples/pipeline_wavernn/wavernn_inference_wrapper.py:164]: I think you can call this method `forward` in the wrapper, and no need to decorate `torch.jit.export`.

**Review by mthrok — APPROVED** [review]
LGTM. The next step will be to move the core inference function to WaveRNN model.

File: examples/pipeline_wavernn/wavernn_inference_wrapper.py (hunk 1 above)
Comment (yangarbiter) [reply examples/pipeline_wavernn/wavernn_inference_wrapper.py:164]: Addressed in #1650

**Pull request merged**